sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2661,4 +2661,186 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
         data.selectExpr("`part.col1`", "`col.1`"))
     }
   }
+
+  test("SPARK-10747: ORDER BY clause with NULLS FIRST|LAST in WINDOW specification") {
```

Comment:
We also need a test for the generic case, e.g.: `SELECT * FROM tbl_a ORDER BY id NULLS LAST`